extension.js

```diff
@@ -567,33 +603,37 @@ const EasyScreenCast_Indicator = new Lang.Class({
             }
 
             //execute post-command
-            if (Settings.getOption('b', Settings.ACTIVE_POST_CMD_SETTING_KEY)) {
-                Lib.TalkativeLog('-*-execute post command');
-
-                //launch cmd after registration
-                var tmpCmd = '/usr/bin/sh -c "' +
-                    Settings.getOption('s', Settings.POST_CMD_SETTING_KEY) + '"';
-
-                var mapObj = {
-                    _fpath: pathFile,
-                    _dirpath: pathFile.substr(0, pathFile.lastIndexOf('/')),
-                    _fname: pathFile.substr(pathFile.lastIndexOf('/') + 1,
-                        pathFile.length)
-                };
+            this._doPostCommand();
+        }
 
-                var Cmd = tmpCmd.replace(/_fpath|_dirpath|_fname/gi,
-                    function(match) {
-                        return mapObj[match];
-                    });
+        this.refreshIndicator(false);
+    },
 
-                Lib.TalkativeLog('-*-post command:' + Cmd);
+    _doPostCommand :function(){
+        if (Settings.getOption('b', Settings.ACTIVE_POST_CMD_SETTING_KEY)) {
+            Lib.TalkativeLog('-*-execute post command');
 
-                //execute post command
-                this.CtrlExe.Spawn(Cmd);
-            }
-        }
+            //launch cmd after registration
+            var tmpCmd = '/usr/bin/sh -c "' +
```

Comment:
The 3 "var" in this block could all be "const" because they are never changed.

Reply:
fixed